src/createServer.js

```diff
@@ -1,8 +1,51 @@
 'use strict';
 
+const fs = require('fs');
+const path = require('path');
+const http = require('http');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', async (req, res) => {
+    const url = new URL(req.url, `http://${req.headers.host}`);
+
+    if (req.method === 'POST' && url.pathname === '/add-expense') {
+      const chunks = [];
+
+      for await (const chunk of req) {
+        chunks.push(chunk);
+      }
+
+      const text = Buffer.concat(chunks).toString();
+      const data = JSON.parse(text);
+      const { date, title, amount } = data;
+
+      if (!date || !title || !amount) {
+        res.statusCode = 400;
+
+        return res.end('All the field must be field');
```

Comment:
There is a typo in the error message: 'All the field must be field'. It should be 'All fields must be filled'.